diopter/compiler.py

```diff
@@ -1394,15 +1420,28 @@ def parse_compilation_setting_from_string(
     )
     CPP_EXT = (".cc", ".cxx", ".cpp")
     C_EXT = (".c",)
+    OBJ_EXT = (".o",)
 
-    sources: list[str] = []
+    sources: list[SourceProgram | ObjectCompilationOutput] = []
     flags: list[str] = []
     for arg in rest:
-        if arg.lower().endswith(CPP_EXT + C_EXT):
-            sources.append(arg)
+        if arg.lower().endswith(CPP_EXT):
+            sources.append(SourceProgram(language=Language.CPP, filename=Path(arg)))
+        elif arg.lower().endswith(C_EXT):
+            sources.append(SourceProgram(language=Language.C, filename=Path(arg)))
+        elif arg.lower().endswith(OBJ_EXT):
+            sources.append(ObjectCompilationOutput(filename=Path(arg)))
         else:
             flags.append(arg)
 
+    # append flags caputred by the parser
+    if parsed_args.MT:
+        for mt in parsed_args.MT:
+            flags.extend(["-MT", mt])
+    if parsed_args.MQ:
+        for mq in parsed_args.MQ:
+            flags.extend(["-MQ", mq])
+
```

Comment:
How is this different than not parsing them? They are anyways appended to the flags. 

I think the solution would be to provide someway in the `CompilationCommand` to emit "fixed" dep files